Tweak default SDFGI settings for better quality

Calinou opened the pull request:
- Enable Read Sky Light to get proper outdoors lighting out of the box.
- Set bounce feedback to 0.5 by default to get a better quality result.
  - [Higher values may cause infinite feedback with bright surfaces.](https://github.com/godotengine/godot/issues/50773#issuecomment-886047017)
- Increase the number of frames to converge to improve quality at the cost of latency. Most scenes are fairly static after all.
- Use 75% Y scale by default as most scenes are not highly vertical.
- Reorder the Y scale enum to go from the lowest Y scale to the highest. Also rename the "Disabled" setting to "100%" for clarity.

**Edit:** This PR was amended in May 2021 to not tweak any performance-related settings. Therefore, performance should be roughly identical to what it was before. Performance tweaks were moved to https://github.com/godotengine/godot/pull/39965.

aaronfranke commented on scene/resources/environment.cpp:
```diff
@@ -1309,16 +1309,16 @@ Environment::Environment() {
 	sdfgi_normal_bias = 1.1;
 	sdfgi_probe_bias = 1.1;
-	sdfgi_y_scale = SDFGI_Y_SCALE_DISABLED;
+	sdfgi_y_scale = SDFGI_Y_SCALE_75_PERCENT;
```
Why default to 75%?

Calinou replied:
I discussed this with @reduz. Most level layouts aren't that vertical, and therefore benefit from having a slightly "flattened" probe structure. This should make leaks less noticeable.

We could go as far as defaulting to 50%, but 75% is probably a safer compromise.

akien-mga commented:
Needs rebase after #40016.

clayjohn reviewed (approved):
This will still need reduz' final approval, but it looks fine.

RaTcHeT302 commented:
i don't think this commit will help at all, there's an issue with the algorithm itself at it's core

Calinou commented:
@RaTcHeT302 The new settings definitely help in non-pathological situations. It's just that you apparently ran into a bug with SDFGI.

aaronfranke commented:
@Calinou What's the status of this? It seems that this is desired, but this needs a rebase before it can be reviewed and merged.

Calinou commented:
@aaronfranke reduz wasn't convinced by all of these default setting changes. I need to discuss this with him again first.

Calinou committed:
Tweak default SDFGI settings for better quality

- Enable Read Sky Light to get proper outdoors lighting out of the box.
- Set bounce feedback to 0.5 by default to get a better quality result.
  - Higher values may cause infinite feedback with bright surfaces.
- Increase the number of frames to converge to improve quality
  at the cost of latency. Most scenes are fairly static after all.
- Use 75% Y scale by default as most scenes are not highly vertical.

- Reorder the Y scale enum to go from the lowest Y scale to the highest.
  Also rename the "Disabled" setting to "100%" for clarity.

reduz commented:
SDFGI needs further work on these areas, will work on it after Alpha

Calinou commented:
> SDFGI needs further work on these areas, will work on it after Alpha

Shouldn't we enable **Read Sky Light** by default at least in the meantime? Right now, the SDFGI appearance on outdoor scenes isn't very appealing until you enable that property.

FilipLundby commented:
> Shouldn't we enable **Read Sky Light** by default at least in the meantime? Right now, the SDFGI appearance on outdoor scenes isn't very appealing until you enable that property.

It would be nice if it was possible to control the shadow color. When **Read sky light** is turned on shadows contains a lot of blue (see image 1). Turning **Read sky light** off makes shadows go black (see image 2).

I am aiming for a more neutral look, like image 3. **I would expect the shadow color of the DirectionalLight to be the shadow color**, but it doesn't seem to do anything. 

The argument might be that the blueish color is more realistic and this might be true, but as a game developer I'm not always going for realism. For winter scene this would be perfect, but for warm summer day it feels out of place. 

Just my two cents, other then that, it looks amazing! :)  

**1. Blue sky and Read sky light on**
![image](https://user-images.githubusercontent.com/9482792/137620188-7d33cd67-1510-4e84-a1cd-e24143824f5e.png)

**2. Read sky light off**
![image](https://user-images.githubusercontent.com/9482792/137620184-9fadc359-c713-4a10-b5c0-c343d0facbc0.png)

**3. Gray sky and Read sky light on**
![image](https://user-images.githubusercontent.com/9482792/137620173-745db57e-a1b6-42a2-ae4c-9c26dd8ea136.png)

**What the scene and sky looks like**
![image](https://user-images.githubusercontent.com/9482792/137620160-9c229d2e-f565-40a5-9eae-4e7ad792c5fa.png)

Calinou commented:
> It would be nice if it was possible to control the shadow color. When Read sky light is turned on shadows contains a lot of blue (see image 1). Turning Read sky light off makes shadows go black (see image 2).

You don't adjust the shadow color. Instead, you adjust the ambient light color. To do this, reduce **Sky Contribution** in the Environment settings (and optionally set a plain color instead of sourcing the sky for ambient light).

Calinou commented:
> Thank you for the feedback @Calinou. I am probably missing something, because your description sounds exactly as how I set up the environment. Ambient color is set to neutral gray and ambient sky contribution is set to 0.

This turned out to be a (likely) bug: https://github.com/godotengine/godot/issues/54681

akien-mga reviewed (approved):
Approved in PR review meeting.
